### PR TITLE
Update headers and add license checker for Flow 23.5

### PR DESCRIPTION
**Pull request by mcollovati**
**DO NOT SQUASH**: headers update and license checker addition are done in two separated commits

### CODE REVIEW

File: flow-client/src/main/frontend/index.ts [hunk 1]
```diff
@@ -1 +1,10 @@
+///
+/// Copyright (C) 2000-2024 Vaadin Ltd
+///
+/// This program is available under Vaadin Commercial License and Service Terms.
+///
+/// See <https://vaadin.com/commercial-license-and-service-terms> for the full
+/// license.
+///
```
Comment (tepi) [inline flow-client/src/main/frontend/index.ts:8]: Not going to check all files, but is there a reason `.ts` files have `///` instead of `/* */`

Comment (mcollovati) [reply flow-client/src/main/frontend/index.ts:8]: The updates are done by license-maven-plugin, but I could not find the reason triple slash is used for `.ts` files 

https://github.com/mathieucarbou/license-maven-plugin/blob/master/license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/document/DocumentType.java#L96C3-L96C42

The plugin allows custom mapping, so I suppose we can change it to `/* */` for typescript files

Comment (tepi) [reply flow-client/src/main/frontend/index.ts:8]: Alright, maybe it should not be changed then. I am not sure what the convention is for `.ts` files. Just caught my eye that it's a bit weird looking.

Comment (mcollovati) [reply flow-client/src/main/frontend/index.ts:8]: I'm not sure the triple slash is correct: https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html

Comment (tepi) [reply flow-client/src/main/frontend/index.ts:8]: Definitely looks not correct to use them for header

**Review by tepi — APPROVED** [review]

File: flow-client/src/main/frontend/index.ts (hunk 1 above)
Comment (mcollovati) [reply flow-client/src/main/frontend/index.ts:16]: Now the headers are duplicate :joy: 
Let me clean up and push the changes

Comment (tepi) [reply flow-client/src/main/frontend/index.ts:16]: Right 😆 Wasn't that obvious the commit..

Comment (mcollovati) [reply flow-client/src/main/frontend/index.ts]: Now it should be fixed

**Review by tepi — APPROVED** [review]

**Pull request merged**